src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -29,6 +29,26 @@
 
 logger = _logging.get_az_logger(__name__)
 
+def _resource_client_factory():
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.resource.resources import ResourceManagementClient
+    return get_mgmt_service_client(ResourceManagementClient)
+
+def cf_providers():
+    return _resource_client_factory().providers
+
+def register_providers():
+    providers = cf_providers()
+
+    namespaces = ['Microsoft.Network', 'Microsoft.Compute', 'Microsoft.Storage']
+    for namespace in namespaces:
+        state = providers.get(resource_provider_namespace=namespace)
+        if state.registration_state != 'Registered':
+            logger.info('registering {}', namespace)
+            providers.register(resource_provider_namespace=namespace)
```

Comment:
Please note, this will only flag the state to 'registering', but it is unknown when it will become "registered". So what is being done here is not reliable, though it is useful